src/logic/map_objects/tribes/production_program.cc

```diff
@@ -2089,7 +2089,7 @@ void ProductionProgram::ActConstruct::execute(Game& game, ProductionSite& psite)
 	Area<FCoords> area(map.get_fcoords(psite.get_position()), radius);
 	if (map.find_reachable_immovables(game, area, &immovables, cstep, FindImmovableByDescr(descr)) !=
 	    0u) {
-		state.objvar = immovables[0].object;
+		state.objvar = immovables.at(game.logic_rand() % immovables.size()).object;
```

Comment:
Looks OK and it's trivial. I wonder though whether it would make sense to skip the randomisation when `immovables.size() == 1`, or it would only make it more complicated.

Reply:
I think that would make it a lot harder to read since the result is always 0 in either case.